tests/lib/rules/accessor-pairs.js

```diff
@@ -46,38 +324,373 @@ ruleTester.run("accessor-pairs", rule, {
+        {
+            code: "var o = { get a() {}, get b() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError, setterError]
+        },
+        {
+            code: "var o = { set a(foo) {}, set b(bar) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError, getterError]
+        },
+        {
+            code: "var o = { get a() {}, set b(foo) {}, set c(foo) {}, get d() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError, getterError, getterError, setterError]
+        },
+
+        // Checks per literal
+        {
+            code: "var o1 = { get a() {} }, o2 = { set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError, getterError]
+        },
+        {
+            code: "var o1 = { set a(foo) {} }, o2 = { get a() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError, setterError]
+        },
+
+        // Combinations or valid and invalid
+        {
+            code: "var o = { get a() {}, get b() {}, set b(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError]
+        },
+        {
+            code: "var o = { get b() {}, get a() {}, set b(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError]
+        },
+        {
+            code: "var o = { get b() {}, set b(foo) {}, get a() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [setterError]
+        },
+        {
+            code: "var o = { set a(foo) {}, get b() {}, set b(bar) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError]
+        },
+        {
+            code: "var o = { get b() {}, set a(foo) {}, set b(bar) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError]
+        },
+        {
+            code: "var o = { get b() {}, set b(bar) {}, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError]
+        },
+        {
+            code: "var o = { get v1() {}, set i1(foo) {}, get v2() {}, set v2(bar) {}, get i2() {}, set v1(baz) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            errors: [getterError, setterError]
+        },
+
+        // In the case of duplicates which don't have the other kind, all nodes are reported
+        {
+            code: "var o = { get a() {}, get a() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [setterError, setterError]
```

Comment:
I think it would be useful to include location data in these errors, to confirm that the same node is not reported twice. Could you please add location data to this test and the one below?

Reply:
This is a very good point, it would be very useful indeed. I'll fix this, just to see first what other changes should be made.

Reply:
This is also done :)